fluss-common/src/test/java/org/apache/fluss/row/GenericMapTest.java

```diff
@@ -0,0 +1,357 @@
+        javaMap.put("c", 3);
+
+        GenericMap genericMap = new GenericMap(javaMap);
+
+        assertThat(genericMap.get("a")).isEqualTo(1);
+        assertThat(genericMap.get("b")).isEqualTo(2);
+        assertThat(genericMap.get("c")).isEqualTo(3);
+        assertThat(genericMap.get("d")).isNull();
+    }
+
+    @Test
+    public void testSize() {
+        Map<String, Integer> javaMap = new HashMap<>();
+        javaMap.put("key1", 100);
+        javaMap.put("key2", 200);
+        javaMap.put("key3", 300);
+        javaMap.put("key4", 400);
+
+        GenericMap genericMap = new GenericMap(javaMap);
+
+        assertThat(genericMap.size()).isEqualTo(4);
+    }
+
+    @Test
+    public void testEmptyMap() {
+        Map<String, Integer> javaMap = new HashMap<>();
+        GenericMap genericMap = new GenericMap(javaMap);
+
+        assertThat(genericMap.size()).isEqualTo(0);
+        assertThat(genericMap.keyArray().size()).isEqualTo(0);
+        assertThat(genericMap.valueArray().size()).isEqualTo(0);
+    }
+
+    @Test
+    public void testKeyArrayAndValueArray() {
+        Map<String, Integer> javaMap = new LinkedHashMap<>();
+        javaMap.put("first", 1);
+        javaMap.put("second", 2);
+        javaMap.put("third", 3);
+
+        GenericMap genericMap = new GenericMap(javaMap);
+
+        InternalArray keyArray = genericMap.keyArray();
+        InternalArray valueArray = genericMap.valueArray();
+
+        assertThat(keyArray.size()).isEqualTo(3);
+        assertThat(valueArray.size()).isEqualTo(3);
+
+        assertThat(keyArray.isNullAt(0)).isFalse();
+        assertThat(valueArray.isNullAt(0)).isFalse();
+    }
+
+    @Test
+    public void testKeyArrayCaching() {
+        Map<String, Integer> javaMap = new HashMap<>();
+        javaMap.put("key", 42);
+
+        GenericMap genericMap = new GenericMap(javaMap);
+
+        InternalArray keyArray1 = genericMap.keyArray();
+        InternalArray keyArray2 = genericMap.keyArray();
+
+        assertThat(keyArray1).isSameAs(keyArray2);
+    }
+
+    @Test
+    public void testValueArrayCaching() {
+        Map<String, Integer> javaMap = new HashMap<>();
+        javaMap.put("key", 42);
+
+        GenericMap genericMap = new GenericMap(javaMap);
+
+        InternalArray valueArray1 = genericMap.valueArray();
+        InternalArray valueArray2 = genericMap.valueArray();
+
+        assertThat(valueArray1).isSameAs(valueArray2);
+    }
+
+    @Test
+    public void testMapWithNullValues() {
+        Map<String, Integer> javaMap = new HashMap<>();
+        javaMap.put("key1", 1);
+        javaMap.put("key2", null);
+        javaMap.put("key3", 3);
+
+        GenericMap genericMap = new GenericMap(javaMap);
+
+        assertThat(genericMap.size()).isEqualTo(3);
+        assertThat(genericMap.get("key1")).isEqualTo(1);
+        assertThat(genericMap.get("key2")).isNull();
+        assertThat(genericMap.get("key3")).isEqualTo(3);
+    }
+
+    @Test
+    public void testEquals() {
```

Comment:
remove all the tests about `equals`, `toString` and `hashcode`. Tests don't need to cover these logics. If we want to improve coverage, they should has already been invoked in other tests and thus has test coverage.